app/controllers/nonprofits_controller.rb

```diff
@@ -55,10 +55,18 @@ def create
   end
 
   def update
-    flash[:notice] = "Update successful!"
-    current_nonprofit.update_attributes params[:nonprofit].except(:verification_status)
-    current_nonprofit.clear_cache
-    json_saved current_nonprofit
+    @form = NonprofitSettingsForm.new(
+      nonprofit: current_nonprofit,
+      attributes: params[:nonprofit].except(:verification_status)
+    )
+
+    if @form.save
+      flash[:notice] = "Update successful!"
+      current_nonprofit.clear_cache
+      json_saved current_nonprofit
+    else
+      json_saved current_nonprofit, 500
```

Comment:
This is not how the `json_saved` method works, I need to change this.

Reply:
Ended up deciding not to use `json_saved` for anything in this method since it's inflexible about status. Used the `json_saved` implementation as a basis and just used render here instead.